cmake/configure/configure_2_trilinos.cmake

```diff
@@ -148,40 +161,124 @@ MACRO(FEATURE_TRILINOS_FIND_EXTERNAL var)
     # with the -std=c++0x flag of GCC, see deal.II FAQ.
     # Test whether that is indeed the case
     #
-    IF(NOT TRILINOS_SUPPORTS_CPP11)
-
-      IF(TRILINOS_HAS_C99_TR1_WORKAROUND)
-        LIST(APPEND TRILINOS_DEFINITIONS "HAS_C99_TR1_CMATH")
-        LIST(APPEND TRILINOS_USER_DEFINITIONS "HAS_C99_TR1_CMATH")
-      ELSE()
-        MESSAGE(STATUS "Could not find a sufficient Trilinos installation: "
-          "The installation is not compatible with the C++ standard selected for "
-          "this compiler."
+    IF(${DEAL_II_TRILINOS_WITH_SACADO})
+      #
+      # Look for Sacado_config.h - we'll query it to determine C++11 support:
+      #
+      DEAL_II_FIND_FILE(SACADO_CONFIG_H Sacado_config.h
+        HINTS ${Trilinos_INCLUDE_DIRS}
+        NO_DEFAULT_PATH NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_PATH
+        NO_SYSTEM_ENVIRONMENT_PATH NO_CMAKE_SYSTEM_PATH NO_CMAKE_FIND_ROOT_PATH
+        )
+      
+      SET(TRILINOS_WITH_MANDATORY_CXX11 FALSE)
```

Comment:
I guess that all of the `TRILINOS_WITH_*` flags defined in `configure_trilinos` should now be immediately defined as `DEAL_II_TRILINOS_WITH_*`? The `TRILINOS_WITH_*` counterparts were previously marked to be cleared at the end of `FindTrilinos` but now I presume they'll remain in CMake's cache (I stand to be corrected on how that mechanism works). Alternatively, we could just [unset](https://stackoverflow.com/a/16309871) these variables manually at the end (i.e. after the `TRILINOS_USER_DEFINITIONS` and `TRILINOS_DEFINITIONS` have been amended.

Reply:
The variables shouldn't be cached. You need to explicitly decide to put a variable in CMake's cache which is not done here.

Reply:
Ok. Thanks for the clarification!

Reply:
Keeping the `TRILINOS_[...]` is fine. Before this PR the `FindTrilinos` module polluted the global namespace with these variables, now the `configure_[...]` module does. No big change :-]